test/jdk/sun/security/pkcs11/Provider/MultipleLogins.java

```diff
@@ -111,23 +125,23 @@ private static void test(SunPKCS11 p) throws Exception {
         try {
             ks.load(null, (char[]) null);
         } catch (IOException e) {
-            if (e.getCause() instanceof LoginException &&
-                    e.getCause().getMessage().contains("No token present")) {
-                // expected
-            } else {
+            if (!(e.getCause() instanceof LoginException) ||
+                !(e.getCause().getMessage().contains("No token present"))) {
+
                 throw new RuntimeException("Token was present", e);
-            }
+            } // else expected
```

Comment:
does this comment mean that the received exception `e` is expected or that you're expecting an `else` clause that isn't here yet?